File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        j = len(heights) -1
        i = 0
        maxArea = 0

        while i <= j:
            width = (j-i)
            height = min(heights[i],heights[j])
            maxArea = max(width * height, maxArea)

            if heights[i] >= heights[j]:
                j-=1
            else:
                i +=1

            
        
        print(maxArea)

        return maxArea




        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
           // starting from the minium index
     int min = prices[0];
     // we could start from prices[0], but we just tryna get the maxiumn number
     int max = 0;

     for(int i = 0; i< prices.length; i++) {
     
     int currindex = prices[i];
      
      min = Math.min(currindex, min);
       
       max = Math.max(max, currindex - min);
     }
     return max;
     }

    
}
